test/extensions/filters/http/ext_authz/ext_authz_test.cc

```diff
@@ -2069,9 +2069,10 @@ TEST_F(HttpFilterTest, EmitDynamicMetadataWhenDenied) {
   EXPECT_EQ("ext_authz_denied", filter_callbacks_.details());
 }
 
-// Verify that when returning an Error response with dynamic_metadata field set, the filter skips
-// dynamic metadata.
-TEST_F(HttpFilterTest, SkipEmittingMetadataWhenError) {
+// Verify that when returning an Error response with dynamic_metadata field set, the filter emits
+// dynamic metadata. In the current implementation, we don't emit metadata. This is here as a
+// reference so if it is needed, emitting dynamic metadata when CheckStatus::Error is possible.
```

Comment:
I think we could simplify this a bit and just say that we'll emit metadata if the ext authz client responds with an error and provides metadata. Leaving this comment here will likely stale out if this changes (who would think to update this?), so I would avoid being too specific here.

Reply:
👍🏽